modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -596,9 +596,17 @@ public CommitData createCommitData() {
 
   @Override
   public synchronized void commit() throws CommitException {
-    SyncCommitObserver sco = new SyncCommitObserver();
-    commitAsync(sco);
-    sco.waitForCommit();
+    SyncCommitObserver sco = null;
+    try {
+      sco = new SyncCommitObserver();
+      commitAsync(sco);
+      sco.waitForCommit();
+    } finally {
+      updates.clear();
+      weakNotification = null;
+      observedColumns.clear();
```

Comment:
The way I found the bug was by commenting out all of the clears except for one and running workerIT.   Sorry for suggesting to clear that, I wasn't thinking about it too much I just looked at all of the instance vars looking for maps and sets.  

I opened #833.